# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: a497ddb030f6
# Full Hash: a497ddb030f6d0852bca56f875b48bc2e1ebc134
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 5: Implement RecvShow r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227690
# ==============================================================================

diff -r 5a94b945a3ad -r a497ddb030f6 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Thu Nov 07 17:32:05 2024 +0000
+++ b/dom/notification/NotificationUtils.h	Thu Nov 07 17:32:05 2024 +0000
@@ -19,6 +19,10 @@
 class Document;
 }  // namespace mozilla::dom
 
+namespace mozilla::dom {
+class IPCNotificationOptions;
+}
+
 namespace mozilla::dom::notification {
 
 /**
@@ -66,6 +70,10 @@
 void ComputeAlertName(nsIPrincipal* aPrincipal, const nsString& aTag,
                       const nsString& aId, nsString& aResult);
 
+nsresult PersistNotification(nsIPrincipal* aPrincipal, const nsString& aId,
+                             const nsString& aAlertName,
+                             const IPCNotificationOptions& aOptions,
+                             const nsString& aScope);
 nsresult UnpersistNotification(nsIPrincipal* aPrincipal, const nsString& aId);
 
 enum class CloseMode {
